webui/react/src/hooks/useSettings.ts

```diff
@@ -251,7 +251,16 @@ const useSettings = <T>(config: SettingsConfig<T>): UseSettingsReturn<T> => {
           const oldSettings = s.get(config.storagePath) ?? {};
           const newSettings = { ...s.get(config.storagePath), ...updates };
           if (!isEqual(oldSettings, newSettings)) {
-            setTimeout(() => userSettings.set(t.any, config.storagePath, newSettings), 0);
+            const props: { [k: string]: t.Type<unknown> } = {};
+            Object.keys(updates).forEach((key) => {
+              // eslint-disable-next-line @typescript-eslint/no-explicit-any
+              const t = (config.settings as any)[key]?.type;
+              if (t !== undefined) {
+                props[key] = t;
+              }
+            });
+            const type = t.type(props);
+            setTimeout(() => userSettings.set(type, config.storagePath, updates), 0);
```

Comment:
Just sending the `updates` here is what I had intended to do. I have to construct the type above because the overload relies on the type to determine whether to use the partiality behavior. Maybe a good argument for it to just be a separate `setPartial` function.